sleap/gui/widgets/video.py

```diff
@@ -1732,6 +1733,211 @@ def updateEdge(self, node: QtNode):
             self.setLine(line)
 
 
+class VisibleBoundingBox(QtWidgets.QGraphicsRectItem):
+    """
+    QGraphicsRectItem for user instance bounding boxes.
+
+    This object defines a scalable bounding box that encases an instance and handles
+    the relevant scaling operations. It is instantiated when its respective QtInstance
+    object is instantiated.
+
+    When instantiated, it creates 4 boxes, which are properties of the overall object,
+    on the corners of the overall bounding box. These corner boxes can be dragged to
+    scale the overall bounding box.
+
+    Args:
+        rect: The :class:`QRectF` object which defines the non-scalable bounding box.
+        parent: The :class:`QtInstance` to encompass.
+
+    """
+
+    def __init__(self, rect, parent=None):
+        super().__init__(rect, parent)
+        if parent is not None:
+            self.box_width = parent.markerRadius
+            color_manager = parent.player.color_manager
+            int_color = color_manager.get_item_color(parent.instance)
+            self.int_color = QColor(*int_color)
+        else:
+            self.box_width = 7
+            self.int_color = Qt.white
+
+        self.parent = parent
+        self.origin = rect.topLeft()
+        self.ref_width = rect.width()
+        self.ref_height = rect.height()
+
+        box_pen = QPen(Qt.black)
+        box_pen.setCosmetic(True)
+        box_brush = QBrush(self.int_color)
+
+        # Create the edge boxes
+        self.top_left_box = QtWidgets.QGraphicsRectItem(parent=self)
+        self.top_left_box.setPen(box_pen)
+        self.top_left_box.setBrush(box_brush)
+        self.top_left_box.setOpacity(0.8)
+        self.top_left_box.setCursor(QCursor(Qt.DragMoveCursor))
+
+        self.top_right_box = QtWidgets.QGraphicsRectItem(parent=self)
+        self.top_right_box.setPen(box_pen)
+        self.top_right_box.setBrush(box_brush)
+        self.top_right_box.setOpacity(0.8)
+        self.top_right_box.setCursor(QCursor(Qt.DragMoveCursor))
+
+        self.bottom_left_box = QtWidgets.QGraphicsRectItem(parent=self)
+        self.bottom_left_box.setPen(box_pen)
+        self.bottom_left_box.setBrush(box_brush)
+        self.bottom_left_box.setOpacity(0.8)
+        self.bottom_left_box.setCursor(QCursor(Qt.DragMoveCursor))
+
+        self.bottom_right_box = QtWidgets.QGraphicsRectItem(parent=self)
+        self.bottom_right_box.setPen(box_pen)
+        self.bottom_right_box.setBrush(box_brush)
+        self.bottom_right_box.setOpacity(0.8)
+        self.bottom_right_box.setCursor(QCursor(Qt.DragMoveCursor))
+
+    def setRect(self, rect: QRectF):
+        # Update edge boxes along with instance box
```

Comment:
Docstring should be written with `"""`:

```suggestion
        """Update edge boxes along with instance box."""
```